### PR TITLE
[Update] DefaultPermissionProvider to expose additional data

### LINKED ISSUE #11: Simplify PersonRole.PopulatePersonPermissions() and ParticipantRole.PopulateParticipantPermissions() by using generated DefaultPermissionProvider class
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-SDK
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

PersonRole.PopulatePersonPermissions() and ParticipantRole.PopulateParticipantPermissions() have overcomplicated implementation that could be substituted by using generated DefaultPermissionProvider class.
### Steps to Reproduce

See the source code CDP4Common/Poco/PersonRole.cs, CDP4Common/Poco/ParticipantRole.cs, CDP4Common/AutoGenHelpers/DefaultPermissionProvider.cs
### System Configuration

- CDP4 version:
  - [x] CDP4Common:         
  - [ ] CDP4JsonSerializer: 
  - [ ] CDP4Dal:            
  - [ ] CDP4JsonFileDal:    
  - [ ] CDP4ServicesDal:    
  - [ ] CDP4WspDal:         
  - [ ] Other:              
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

### PR DESCRIPTION
**Pull request by IevhenIkonnykov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-SDK-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Update DefaultPermissionProvider to expose additional data and ParticipantRole and PersonRole to utilize it. Details see #11 
Closes #11

### CODE REVIEW

File: CDP4Common/Poco/PersonRole.cs [hunk 1]
```diff
@@ -42,30 +41,16 @@ public partial class PersonRole
...
-                this.PersonPermission.Add(personPermission);
-            }
+            var provider = new DefaultPermissionProvider();
+            provider.GetDefaultTypeNamePersonPermissions()
+                    .Where(x => x.Value.Equals(PersonAccessRightKind.NONE))
+                    .ToList()
+                    .ForEach(x => {
+                        var personPermission = new PersonPermission(Guid.NewGuid(), null, null);
+                        personPermission.AccessRight = x.Value;
+                        personPermission.ObjectClass = (ClassKind)Enum.Parse(ClassKind.GetType(), x.Key);
```
Comment (lxatrhea) [inline CDP4Common/Poco/PersonRole.cs]: Maybe store ClassKind.GetType() in a var out of the ForEach loop, but that's up to you.

File: CDP4Common/Poco/ParticipantRole.cs [hunk 2]
```diff
@@ -41,31 +40,17 @@ public partial class ParticipantRole
...
-            }
+        {
+            var provider = new DefaultPermissionProvider();
+            provider.GetDefaultTypeNameParticipantPermissions()
+                    .Where(x => x.Value.Equals(ParticipantAccessRightKind.NONE))
+                    .ToList()
+                    .ForEach(x => {
+                        var participantPermission = new ParticipantPermission(Guid.NewGuid(), null, null);
+                        participantPermission.AccessRight = x.Value;
+                        participantPermission.ObjectClass = (ClassKind)Enum.Parse(ClassKind.GetType(), x.Key);
```
Comment (lxatrhea) [inline CDP4Common/Poco/ParticipantRole.cs]: Maybe store ClassKind.GetType() in a var out of the ForEach loop, but that's up to you.

File: CDP4Common/Poco/PersonRole.cs (hunk 1 above)
Comment (IevhenIkonnykov) [reply CDP4Common/Poco/PersonRole.cs]: No problem

File: CDP4Common/Poco/ParticipantRole.cs (hunk 2 above)
Comment (IevhenIkonnykov) [reply CDP4Common/Poco/ParticipantRole.cs]: No problem

**Pull request merged**